crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1248,6 +1288,7 @@ impl<'db> Type<'db> {
             | Type::SliceLiteral(_)
             | Type::Tuple(_)
             | Type::LiteralString => Type::Unknown,
+            Type::Type(..) => Type::Unknown,
```

Comment:
I didn't include it in the block above, as the comment would not be correct for `Type::Type`. But I also don't know how one would turn `type[C]` into an instance of some kind.

Reply:
You can call a runtime object with type `type[C]` at runtime to create a runtime object of type `C`: https://mypy-play.net/?mypy=latest&python=3.12&gist=8961e1a985e66a0022bd68baf514db9d

I think this should be the same as the `ClassLiteral` branch here, i.e.

```suggestion
            Type::Type(ClassLiteralType { class }) => Type::anonymous_instance(*class),
```